design/endpoint-slice-support.md

```diff
@@ -0,0 +1,101 @@
+# Contour k8s endpoint slice support
+
+Contour currently supports Kubernetes endpoints.
+This document proposes a process to enable support for k8s endpoint slices and, eventually, migrate to exclusively using it.
+
+## Goals
+
+- Contour will support k8s endpoint slices.
+- The endpoint slices will be gated by a feature flag in the Contour config.
+- Kubernetes endpoints will still be the default mode of operation.
```

Comment:
IMO, once we're confident in the EndpointSlices implementation, we should move to using it as the default, and eventually remove the Endpoint option/code altogether. Let's call that out here as well (assuming folks agree) so it's clear what the plan is.

Reply:
added a small migration section in https://github.com/projectcontour/contour/pull/5703/commits/63fac731b8b745fbe893ca69e3739cd3b21641f5